README.md

```diff
@@ -2,208 +2,394 @@
+variable.
+
+This example demonstrates two of the three different levels of scope in
+JavaScript. After the next few sections, you should be able to come back to this
+example and determine which two.
 
-**_Top Tip_**: If a variable or function is **not** declared inside a function
-or block, it's in the global execution context.
+Let's cover those three scope levels now.
 
-## Function scope
+## Global Scope
 
-When we declare a new function and write some code in the function body, we're
-no longer in the global execution context. The function creates a new execution
-context with its own scope. Inside the function body, we can reference variables
-and functions declared in the function's scope:
+Global scope is the **outermost scope**. Anything defined here is accessible
+everywhere else in your code. This makes it the simplest scope to utilize as it
+does not create any limitations on where you can access variables.
+
+Anything defined at the top level of a file is considered to be definied
+globally. In other words, **anything defined _outside_ of curly braces** `{}`
+are globally scoped.
+
+Let's take a look at some examples. Imagine the following is _all_ the code
+inside a file:
 
 ```js
-function myFunc() {
-  const myVar = 42;
+const accessibleAnywhere = true;
 
-  return myVar * 2;
+function globalFn() {
+  return true;
 }
-// => undefined
 
-myFunc();
-// => 84
+console.log(accessibleAnywhere); // => true
+globalFn(); // => true
 ```
 
-However, from outside the function, we can't reference anything declared inside
-of it:
+Both the variable and function are defined at the top level of the file, making
+them both available in the global scope. If we were to continue adding code to
+this file, we could access both of them anywhere we want.
+
+While global scope makes it easy to access variables and functions wherever and
+whenever we want to, it is not always a good idea. In general, it's best
+practice to make variables and functions available only where they're needed
+— and nowhere else.
+
+Making a variable available in places that shouldn't have access to it can lead
+to bad things and make your debugging process more complex. The more pieces of
+code that can access a given variable, the more places you have to check for
+bugs if/when that variable contains an unexpected value.
+
+This is where the other levels of scope comes in handy.
+
+## Function Scope
+
+Function scope is the **scope within a single function**. Anything **defined
+inside the body of a function** is considered to be part of that function's
+scope.
+
+Anything defined in the function's scope can be accessed anywhere _within the
+function's body_. For example:
 
 ```js
-function myFunc() {
-  const myVar = 42;
+let count = 0;
+
+function counter() {
+  const resetValue = 100;
+  console.log("Counter will reset after: ", resetValue);
+
+  if (count < resetValue) {
+    count++;
+  } else {
+    count = 0;
+  }
```

Comment:
I wonder if this example will confuse students since it will only increment once and the count won't be reset. I'm concerned that trying to figure out what it does (or doesn't do) may distract them from the main point

Reply:
I think that's a good point, Liz.